### PR TITLE
Added depth and stencil support for Amber Script

### PR DESCRIPTION
**Pull request by asuonpaa**
Fixes https://github.com/google/amber/issues/826

### CODE REVIEW

File: docs/amber_script.md [hunk 1]
```diff
@@ -368,6 +368,57 @@ The following commands are all specified within the `PIPELINE` command.
...
+```groovy
+  # Set stencil test settings. |face| can be front, back, or front_and_back.
+  # |test_enable| is either on or off and affects both faces. |fail_op|, |pass_op|,
+  # and |depth_fail_op| are selected from the stencil operations table above,
+  # and |compare_op| from the compare operations table. |compare_mask|, |write_mask|,
+  # and |reference| are 8bit unsigned integer values (range 0..255).
+  STENCIL {face}
+    TEST {test_enable}
+    FAIL {fail_op}
+    PASS {pass_op}
```
Comment (dj2) [inline docs/amber_script.md:413]: Let's call these `FAIL_OP` and `PASS_OP`

File: docs/amber_script.md [hunk 2]
```diff
@@ -368,6 +368,57 @@ The following commands are all specified within the `PIPELINE` command.
...
+ * `always`
+
+```groovy
+  # Set depth test settings. All enable options are specified with keywords on and off.
+  # BOUNDS and BIAS values are specified with decimal numbers. |compare_op| is selected
+  # from the list of compare operations above.
+  DEPTH
+    TEST {test_enable}
+    WRITE {write_enable}
+    COMPARE {compare_op}
```
Comment (dj2) [inline docs/amber_script.md:388]: Let's call it `COMPARE_OP`

File: docs/amber_script.md [hunk 3]
```diff
@@ -368,6 +368,57 @@ The following commands are all specified within the `PIPELINE` command.
...
+  # |test_enable| is either on or off and affects both faces. |fail_op|, |pass_op|,
+  # and |depth_fail_op| are selected from the stencil operations table above,
+  # and |compare_op| from the compare operations table. |compare_mask|, |write_mask|,
+  # and |reference| are 8bit unsigned integer values (range 0..255).
+  STENCIL {face}
+    TEST {test_enable}
+    FAIL {fail_op}
+    PASS {pass_op}
+    DEPTH_FAIL {depth_fail_op}
+    COMPARE {compare_op}
```
Comment (dj2) [inline docs/amber_script.md:415]: `COMPARE_OP`

**Comment by asuonpaa** [comment]
Do we want to split this into several PRs, or should I do a new branch and add those as separate commits and then force push to this one?

File: docs/amber_script.md (hunk 1 above)
Comment (asuonpaa) [reply docs/amber_script.md:413]: I would then also rename DEPTH_FAIL to DEPTH_FAIL_OP.

Comment (dj2) [reply docs/amber_script.md:413]: Nice, catch, yea that one too.

**Comment by dj2** [comment]
I'd prefer separate PRs.

**Comment by dj2** [comment]
Can this be closed now? I believe all the parts have landed?

**Pull request closed without merging**